packages/provider/src/relayer/local-relayer.ts

```diff
@@ -23,6 +25,40 @@ export class LocalRelayer extends BaseRelayer implements IRelayer {
     )
   }
 
+  async estimateGasLimits(
+    config: ArcadeumWalletConfig,
+    context: ArcadeumContext,
+    ...transactions: ArcadeumTransaction[]
+  ): Promise<ArcadeumTransaction[]> {
+    const walletAddr = addressOf(config, context)
+
+    const gasCosts = await Promise.all(transactions.map(async (t) => {
+      // Fee can't be estimated locally for delegateCalls
+      if (t.delegateCall) {
+        return DEFAULT_FEE
+      }
+
+      // Fee can't be estimated for self-called if wallet hasn't been deployed
```

Comment:
how come we can't estimate a self-called wallet that hasn't been deployed?

Reply:
If the wallet is a self-call we need to have the wallet code deployed, otherwise the estimation call will be performed to a wallet without code and no "real" gas estimation will take place.